### DISCUSSION TITLE
Missing samplers in SDXL?

### DISCUSSION THREAD

**Discussion by mkossigit** [vladmandic/sdnext#2116] [Q&A] [answered]
Not sure if this is normal but when switching to Stable Diffusion XL with Diffusers backend, many samplers are missing.
Only ones i see are:
![samplers](https://github.com/vladmandic/automatic/assets/90536087/3b0c8c5d-f379-4214-beeb-2a924619357c)

Switching back to SD, original backend,  shows all of them again.

I did a search and found one response in terms of checking in sampler settings, but they do not show up there is SDXL mode, and everything else is ticked to show.

**Comment by vladmandic** [comment, marked as answer]
if you're talking about *SDE or *Karras (for example), those are not samplers (they never were), those are **settings** applied to samplers.
so check settings -> samplers and you can set or unset those.

diffusers mode received this change, same change will be done to original backend as well.

**Reply by mkossigit** [reply]
Thx! One of the reasons i asked and was confusing was when copying and pasting SDXL civitai examples, using the Read Generation button would change the sampler field to "DPM++ 3M SDE Karras" or whatever sampler, even though it wont appear for me on the list.

**Reply by gohan2091** [reply]
Yes I am wondering this too, if I load an old image it will change the sampler box to DPM++ 3M SDE Karras even though it's not on the list

**Comment by Tormound** [comment]
Speaking of missing samplers was the Restart sampler ever added and i just can't find it or is yet to be added?

**Reply by vladmandic** [reply]
yes to be added. soon.

**Comment by gohan2091** [comment]
I'm still confused. How do I get DPM++ 3M SDE Karras? In my attached screenshot, I have Karras ticked so the DPM++ 2M I have selected would turn into "DPM++ 2M Karras" correct? so how do I get the SDE part? and how do I get 3M and not 2M?
![Screenshot 2024-01-02 041652](https://github.com/vladmandic/automatic/assets/127342824/66f84cf6-726d-4f1d-9bbe-77639c475fb5) also I don't know what "Low Order" and "Dynamic Threshold" is. Any idea?

**Reply by lbeltrame** [reply]
You'll find the SDE sampler in the Settings (Sampler Options I believe, but I can't check now), tick it (it's likely unticked), apply settings, then restart SD.Next.

**Reply by brknsoul** [reply]
3M isn't listed in Sampler Settings.

**Reply by lbeltrame** [reply]
This is what I have on a freshly updated SD.Next:

![immagine](https://github.com/vladmandic/automatic/assets/25192/799aca6d-ab6c-40a9-8ba4-6b157b64e507)

**Reply by gohan2091** [reply]
Is that the original backend? I am asking in regards to the diffusers backend. This is what I have listed in my settings:
![Screenshot 2024-01-02 161428](https://github.com/vladmandic/automatic/assets/127342824/22ad2f8d-912a-468c-b7de-04914756001b)

**Reply by lbeltrame** [reply]
Probably. I admit haven't checked that list under diffusers in a while (most of the SDXL models I use didn't recommend that sampler specifically).

**Reply by vladmandic** [reply]
- SDE is a solver type, select it via settings -> samplers -> dpm solver algorithm.
- DPM 3M does not yet exist in diffusers backend. benefits over 2M are near-trivial.
